### ISSUE TITLE
Feature: Markdown for Alerts or Note blocks and labels

### ISSUE THREAD

**Issue by awidjaja** [dendronhq/dendron#1804]
## Please select if your request is either something new or an enhancement ##

- [ ] Enhancement of an existing Feature.
- [x] Request of a new feature.

## Please select the area your request applies to. (Multiple selections are possible) ##

- [ ] Onboard. Initial vault setup and import/export pods
- [x] Create. Note creation, lookup, snippets and templates
- [ ] Retrieve. Backlinks, references, graph view
- [ ] Structure. Refactoring, multi-vault and schemas
- [x] Publish. Sharing your repo with the world
- [ ] Misc (Choose this if your not sure)

## Is your feature request related to a problem? Please describe ##

People using Microsoft and/or Adobe docs authoring tools won't be able to migrate to Dendron. 
Why? Markdown created using Microsoft and adobe authoring tool will include some markdown extension called `alerts`  or note blocks with predefined `[!TAG]` format. 

Microsoft: 

- Note, Tip, Important, Caution, Warning

![image](https://user-images.githubusercontent.com/1708624/143793701-340015dd-fb27-4b18-be49-9f65aa6452d3.png)

Adobe: 

- NOTE, TIP, IMPORTANT, CAUTION, WARNING, ADMINISTRATION, AVAILABILITY, PREREQUISITES.

![image](https://user-images.githubusercontent.com/1708624/143793766-aaf4b016-cf3a-449e-a3c5-f5a625db916b.png)

It is important for people who want to publish documentations for software or some tutorials.

## Describe the solution you'd like ##

Dendron should recognize [!TAG] format and render inside a prominent box, with certain color and icon. For the color, provide a way for people to edit the css with the required color and icon for each TAG they use. Keeping the same markdown will allow people to migrate the docs without extra work to replace the markdown.

Additionaly, people using gitlab issues are familiar with adding inline labels to provide easy way to tag the documents for easy grouping and text search. Dendron should recognize the `~"tag::tag"` format and render it as a thin badge with the certain background color just like in gitlab. 

![image](https://user-images.githubusercontent.com/1708624/143793906-7c4ca76e-89e4-485e-8ac7-70036f70120a.png)

Both markdown tags will not cause problem with the standard markdown renderer. The alert tag would not be shown and the label tag will just be rendered as is. If it is not desire, the label format can also follow the tag format, e.g. [~tag:tag] for inline labels, and [!~tag::tag] for alert/note blocks with labels .

Related snippets should also be added to ease editing.

With above additions, Dendron can be use by people with serious need for markdown based publishing. 

## Describe alternatives you've considered ##

Keep using Microsoft or Adobe docs authoring tool.

**Comment by jonathanyeung** [comment]
Thank you @awidjaja for the suggestion. We're continually working to improve our markdown rendering behavior.

**Comment by ScriptAutomate** [comment]
Hmmm could potentially combine this issue with https://github.com/dendronhq/dendron/issues/1287

**Comment by awidjaja** [comment]
Yes, it is generally similar use case. Dendron should choose one standard markdown notation for alerts and label and automatically converts other known variations during pod import and then expose a standardized css class that can be customized.  I like the idea of using block quote notation which provide more flexibility for longer or multi paragraph as well as indentation, maybe combined with tag/label notation. e.g.

>!Tip
>
>This is a standard tip. 

>!Warning 
>
>This is a warning ~"status::open". 

It will be rendered naturally in a standard markdown too without the icon and background color. As long as there is auto conversion tool during import.
